pTables.py

```diff
@@ -0,0 +1,235 @@
+            print(self._t)
+
+
+    def get_string_version(self):
+        """
+        :return: The string variable of the table representation.
+        """
+        if not self.is_table_empty():
+            return self._t.get_string()
+
+
+    def sort_table(self, s):
+        """
+        function checks whether or not the string passed is in the header of the table or not. If it is not,
+        the function does nothing.
+        :param s: String passed by the caller. If not a string when classed, function converts (s) to a string.
+        :return: the string representation of the table object, sorted by the string passed by caller.
+        """
+        if not self.is_table_empty():
+            s = str(s)
+            if s in self.header:
+                return self._t.get_string(sortBy=s)
+
+
+
+########################################################################################################################
+########################################################################################################################
+############################# Private functions to class: MyTable    Please No Touch ###################################
+########################################################################################################################
+########################################################################################################################
+
+    def _get_data_row_length(self):
+        """
+        Private function to class MyTable.
+
+        Function assigns the member variable self.dataRowLength with the length of all rows in the table
+        containing the data. This length excludes the header of the table, self.matrixTable[0]
+        """
+
+        if not self.is_table_empty():
+            self.dataRowLength = len(self.matrixTable[1:])
+
+
+    def _put_data(self):
+
+        """
+         Private function to class MyTable.
+         Function installs the data in the table passed to this table objects constructor.
+        :return:
+        """
+        if not self.is_table_empty():
+            self._t.field_names = self.header = self.matrixTable[0]
+            [self._t.add_row(i) for i in self.matrixTable[1:]]
+            self._get_max_column_lengths()
+
+    def _get_max_column_lengths(self):
+        """
+         Private function to class MyTable.
+         Function loops through an calculates the maximum length of each column in the table. It does this by
+         finding the longest string present in any column in the table. Then installs this value in the member variable
+         self.columnWidth array. Where each number in the array corresponds to that columns max width.
+        :return:
+        """
+        if not self.is_table_empty():
+            for i in range(len(self.matrixTable[0])):
+                self.columnWidth.append(max([sub[i] for sub in [[len(word) for word in index] for index in self.matrixTable[:]]]))
```

Comment:
This line is really hard to read and understand what it is doing.

Reply:
ok

Reply:
line 24 is looping through the size of the header row in the table. So however long the first nested  list is, in self.matrixTable.  Therefore at line 124, it is telling line 125. I want you to loop through and find me the longest string in each column. Essentially line 125 loops through one column looking for the MAX length of a string. Line 124 is telling 125 how many columns to loop through.

Reply:
Once it find the longest string in a column. Line 125 appends that value to the corresponding cell in the class member variable self.columnWidth.  That way each integer value in each cell in this list, corresponds to the max length each column should be in the table.